res/controllers/Keyboard-Default.kbd.xml

```diff
@@ -0,0 +1,382 @@
+<?xml version="1.0" ?>
+<MixxxKeyboardPreset mixxxVersion="2.0.1+" schemaVersion="1">
+  <info>
+    <name>Default Mapping</name>
+    <author>Mixxx</author>
+    <description>Default mapping for Mixxx.</description>
+  </info>
+  <controller>
+    <group name="[AutoDJ]">
+      <control action="enabled">
+        <keyseq final="1">Shift+F12</keyseq>
+      </control>
+      <control action="fade_now">
+        <keyseq final="1">Shift+F11</keyseq>
+      </control>
+      <control action="shuffle_playlist">
+        <keyseq final="1">Shift+F9</keyseq>
+      </control>
+      <control action="skip_next">
+        <keyseq final="1">Shift+F10</keyseq>
+      </control>
+    </group>
+    <group name="[Channel1]">
+      <control action="LoadSelectedTrack">
+        <keyseq final="1">Shift+Left</keyseq>
+      </control>
+      <control action="back">
+        <keyseq lang="en_US" scancode="30">a</keyseq>
```

Comment:
yes, IMHO you can go for it.

There shouldn't be effectively an issue with duplicated keys, since we do
not receive a scancode from the OS.

The user can use exactly what he receives. If this is ambiguous, he has a
problem anyway. And the current proposal even allows a workaround by using
the lang and the position by tag to express the key by a foreign layout.

Am 21.08.2016 5:30 nachm. schrieb "Jordi Ortolá Ankum" <
notifications@github.com>:

> In res/controllers/Keyboard-Default.kbd.xml
> <https://github.com/mixxxdj/mixxx/pull/966#discussion_r75596934>:
>
> > +      <control action="fade_now">
> > +        <keyseq final="1">Shift+F11</keyseq>
> > +      </control>
> > +      <control action="shuffle_playlist">
> > +        <keyseq final="1">Shift+F9</keyseq>
> > +      </control>
> > +      <control action="skip_next">
> > +        <keyseq final="1">Shift+F10</keyseq>
> > +      </control>
> > +    </group>
> > +    <group name="[Channel1]">
> > +      <control action="LoadSelectedTrack">
> > +        <keyseq final="1">Shift+Left</keyseq>
> > +      </control>
> > +      <control action="back">
> > +        <keyseq lang="en_US" scancode="30">a</keyseq>
>
> However, it looks like it exposes too much internals to the user.
>
> True, although I don't know if that would still be an issue as soon as we
> have the mapping GUI.
>
> Do we need the scancode option?
>
> I am a bit sceptical about only having the possibility to target keys on
> their keysyms *** rather than on their scancodes because there might be
> keys that share the same keysym. To find out if this was a thing, I scanned
> our current layouts
> <https://github.com/Tomasito665/mixxx/blob/develop/src/controllers/keyboard/layouts.cpp>
> for these type of keys that share the same keysym, with the same modifier.
> This was the outcome:
>
> On layout English (US), <AB09> is identical to <LSGT> with shift. Character: '>'
> On layout Greek (Greece), <AC02> is identical to <AD02> with shift. Character: 'Σ'
>
> Which means that we can not do:
>
> <!-- In both cases there are two possible scancodes, based on our layouts data -->
> <keyseq byPositionOf="en_US">Shift+&gt;</keyseq>
> <keyseq byPositionOf="el_GR">Shift+Σ</keyseq>
>
> However, these two are special cases, because:
>
>    1. Keyboards for en_US English (US) layouts usually don't have this
>    105th key.
>    2. Both keys and hold different lower-case sigma's and thus it's
>    logical that they will produce the same upper-case sigma Σ when shifted.
>
> So, I think we can live with it. Though, this does only say something
> about the 16 layouts we currently have in layouts.cpp. So, I don't know
> if this could potentially become a problem if we add more layouts, which do
> have more of these duplicate keys.
>
> What do you think? Shall I just implement it?
>
> ** We do not map on keysyms directly but on their Unicode characters
> <https://github.com/Tomasito665/mixxx/blob/develop/scripts/layouts_tool/utils.h#L15>.*
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/966/files/995266eed0038476b7212100c88a58eb6366b750#r75596934>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABsfIsBFfwLt4qlWbSQ7WFVV0qflgvxbks5qiG8SgaJpZM4I6SSC>
> .
>